src/rabbit_reader.erl

```diff
@@ -387,10 +387,17 @@ start_connection(Parent, HelperSup, Deb, Sock) ->
         %% socket w/o delay before termination.
         rabbit_net:fast_close(RealSocket),
         rabbit_networking:unregister_connection(self()),
-	rabbit_core_metrics:connection_closed(self()),
+        rabbit_core_metrics:connection_closed(self()),
+        ClientProperties = case get(client_properties) of
```

Comment:
Rather than use the process dictionary, could we instead use [`sys:get_status(self())`](http://erlang.org/doc/man/sys.html#get_status-1) to get the current process state and extract `client_properties` from that?

Reply:
`sys:get_status/1` will send a system call message to the process. This doesn't work with `self()`

Reply:
Thanks for the explanation!